src/codeflare_sdk/ray/cluster/generate_yaml.py

```diff
@@ -88,9 +89,14 @@ def update_names(
 
 def update_image(spec, image):
     containers = spec.get("containers")
-    if image != "":
-        for container in containers:
-            container["image"] = image
+    if not image:
+        python_version = f"{sys.version_info.major}.{sys.version_info.minor}"
+        if python_version.startswith("3.9"):
```

Comment:
Can we have some error handling here for if a user (for some reason) uses Py 3.12 or something so that they are reminded only Py 3.xx is supported 
Otherwise this looks good I can test this out soon

Reply:
Maybe have a const like `SUPPORTED_PYTHON_VERSIONS=["3.9", "3.11"]` which could be used to compare WDYT?

Reply:
I've enhanced the function now by using constants, great idea!!